### PR TITLE
[SW-1847] Automatically Document Sparkling Water Algo API

### PR DESCRIPTION
**Pull request by mn-mikke**

### CODE REVIEW

File: doc/src/site/sphinx/ml/sw_xgboost.rst [hunk 1]
```diff
@@ -1,8 +1,8 @@
 Train XGBoost Model in Sparkling Water
 --------------------------------------
 
-Sparkling Water provides API for H2O XGBoost in Scala and Python.
-The following sections describe how to train the XGBoost model in Sparkling Water in both languages.
+Sparkling Water provides API for H2O XGBoost in Scala and Python. The following sections describe how to train
+theXGBoost model in Sparkling Water in both languages. See also :ref:`parameters_H2OXGBoost`.
```
Comment (jakubhava) [inline doc/src/site/sphinx/ml/sw_xgboost.rst:5]: nit: missing space after `the`

File: doc/src/site/sphinx/ml/sw_target_encoder.rst [hunk 2]
```diff
@@ -32,55 +32,6 @@ An example of converting a categorical feature to continues with Target Encoder
 Target Encoding can help to improve accuracy of machine learning algorithms when columns with high
 cardinality are used as features during a training phase.
 
-Parameters
```
Comment (jakubhava) [inline doc/src/site/sphinx/ml/sw_target_encoder.rst:35]: What  do you think of adding link to generated parameters doc?

File: doc/src/site/sphinx/ml/sw_kmeans.rst [hunk 3]
```diff
@@ -102,6 +102,7 @@ The following sections describe how to train the KMeans model in Sparkling Water
 
 H2O KMeans Parameters
 ~~~~~~~~~~~~~~~~~~~~~
+See also :ref:`parameters_H2OKmeans`.
 
 - **maxIterations**
```
Comment (jakubhava) [inline doc/src/site/sphinx/ml/sw_kmeans.rst:107]: Should we keep the parameters listed bellow?

File: doc/src/site/sphinx/ml/sw_xgboost.rst (hunk 1 above)
Comment (mn-mikke) [reply doc/src/site/sphinx/ml/sw_xgboost.rst:5]: will fix, thanks!

File: doc/src/site/sphinx/ml/sw_target_encoder.rst (hunk 2 above)
Comment (mn-mikke) [reply doc/src/site/sphinx/ml/sw_target_encoder.rst:35]: I wanted to do it, but forgot about that. Good catch, thanks!

File: doc/src/site/sphinx/ml/sw_kmeans.rst (hunk 3 above)
Comment (mn-mikke) [reply doc/src/site/sphinx/ml/sw_kmeans.rst:107]: I was also thinking about that but I'm not sure whether it's better remove it or not. The parameter descriptions comes from H2O-3 V3 classes and sometimes give very little pieces of information. On the other hand, the link contains a complete list of parameters.

I will do whatever you prefer...

Comment (jakubhava) [reply doc/src/site/sphinx/ml/sw_kmeans.rst:107]: Let's keep it. Small duplication is better then missing documentation. WDYT?

**Pull request merged**